evap/staff/tests/test_importers.py

```diff
@@ -374,6 +393,113 @@ def test_replace_consecutive_and_trailing_spaces(self):
 
+    def test_existing_course_is_not_recreated(self):
+        self.create_existing_course()
+
+        old_course_count = Course.objects.count()
+        old_dict = model_to_dict(self.existing_course)
+
+        __, warnings, __ = EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertIn(
+            "Course Shake (Schütteln) already exists with identical attributes. Course is not created and users are put into the evaluation of that course.",
+            warnings[ImporterWarning.EXISTS],
+        )
+        expected_course_count = old_course_count + 22
+        self.assertEqual(Course.objects.count(), expected_course_count)
+        self.existing_course.refresh_from_db()
+        self.assertEqual(old_dict, model_to_dict(self.existing_course))
+
+    def test_existing_course_degree_is_added(self):
+        self.create_existing_course()
+
+        # The existing course exactly matches one course in the import data by default. To create a conflict, the degrees are changed
+        self.existing_course.degrees.set([Degree.objects.get(name_de="Master")])
+
+        EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertSetEqual(
+            set(self.existing_course.degrees.all()), set(Degree.objects.filter(name_de__in=["Master", "Bachelor"]))
+        )
+
+    def test_existing_course_users_added_to_evaluation(self):
+        self.create_existing_course()
+
+        self.assertFalse(self.existing_course_evaluation.participants.exists())
+
+        EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertIn(
+            UserProfile.objects.get(email="lucilia.manilium@institution.example.com"),
+            self.existing_course_evaluation.participants.all(),
+        )
+
+    def test_existing_course_different_attributes(self):
+        self.create_existing_course()
+        self.existing_course.type = CourseType.objects.get(name_de="Seminar")
+        self.existing_course.responsibles.set(
+            [baker.make(UserProfile, email="responsible_person@institution.example.com")]
+        )
+        self.existing_course.save()
+
+        old_course_count = Course.objects.count()
+        old_dict = model_to_dict(self.existing_course)
+
+        __, __, errors = EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertIn("the course type does not match", errors[ImporterError.COURSE][0])
+        self.assertIn("the responsibles of the course do not match", errors[ImporterError.COURSE][0])
```

Comment:
What is this `[0]`? To check that this string is among any of the errors, we usually assert that it is in `"".join(errors[ImporterError.COURSE])`